proxy/src/http/server.rs

```diff
@@ -0,0 +1,27 @@
+use anyhow::anyhow;
+use hyper::{Body, Request, Response, StatusCode};
+use std::net::TcpListener;
+use utils::http::{endpoint, error::ApiError, json::json_response, RouterBuilder, RouterService};
+
+async fn status_handler(_: Request<Body>) -> Result<Response<Body>, ApiError> {
+    json_response(StatusCode::OK, "")
+}
+
+fn make_router() -> RouterBuilder<hyper::Body, ApiError> {
+    let router = endpoint::make_router();
+    router.get("/v1/status", status_handler)
+}
+
+pub async fn thread_main(http_listener: TcpListener) -> anyhow::Result<()> {
```

Comment:
Nit: since it is not a thread 
```suggestion
pub async fn task_main(http_listener: TcpListener) -> anyhow::Result<()> {
```

Reply:
This misnomer is out of scope but I get what you're saying, will change in a follow-up PR.